go-controller/pkg/ovn/ovn.go

```diff
@@ -570,6 +584,98 @@ func (oc *Controller) WatchNetworkPolicy() error {
 	return err
 }
 
+// WatchCRD starts the watching of the CRD resource and calls back to the
+// appropriate handler logic
+func (oc *Controller) WatchCRD() error {
+	_, err := oc.watchFactory.AddCRDHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			crd := obj.(*apiextension.CustomResourceDefinition)
+			klog.Errorf("ADDING CRD - %s\n", crd.Name)
+			if crd.Name == egressfirewallCRD {
+				err := oc.watchFactory.EgressFirewallWatchFactory()
+				if err != nil {
+					klog.Errorf("Error Creating EgressFirewallWatchFactory: %v", err)
+					return
+				}
+				h, err := oc.WatchEgressFirewall()
+				if err != nil {
+					klog.Errorf("Error watching CRD - %v", err)
+					return
+				}
+				oc.egressFirewallHandler = h
+
+			}
+		},
+		UpdateFunc: func(old, newer interface{}) {
+		},
+		DeleteFunc: func(obj interface{}) {
+			crd := obj.(*apiextension.CustomResourceDefinition)
+			klog.Errorf("Deleting CRD - %s\n", crd.Name)
+			if crd.Name == egressfirewallCRD {
```

Comment:
@JacobTanenbaum hmm, do we really care if the CRD gets removed? I don't think we do; if you add the CRD and then remove it, that's your problem :)  Unless it starts spewing errors or something?

Reply:
It does start spewing errors. the informer constantly tries to listen to the egressFirewall api object and logging that the apiserver does not have that defined

Reply:
Can we just log loudly and exit if it gets removed and then when the pod gets restarted it will not detect the CRD and all will be well?

Reply:
It does start spewing errors. the informer will try to list and watch the CRD type and constantly report that it does not exist

Reply:
why would that be better then cleaning up after the CRD removal?

Reply:
Fair.  I thought Dan was saying that he didn't like that code.  But if it works, we can use it.

Reply:
resolved in a voice chat and the consciousness was to keep the cleanup code